frontend/app/src/components/settings/data-security/DataManagement.vue

```diff
@@ -0,0 +1,192 @@
+                      icon
+                      :disabled="!!!exchange"
+                      v-bind="attrs"
+                      v-on="on"
+                      @click="confirmExchangePurge = true"
+                    >
+                      <v-icon>fa-trash</v-icon>
+                    </v-btn>
+                  </template>
+                  <span>Purge cached data for the selected exchange</span>
+                </v-tooltip>
+              </v-col>
+            </v-row>
+            <status-button
+              class="data-management__purge-all-exchange"
+              tooltip="Purge all the cached exchange data. (deposits/withdrawals/trades)"
+              :success-message="
+                allExchangePurgeSuccess
+                  ? 'Cached data for all the available exchanges were successfully purged.'
+                  : null
+              "
+              :error-message="
+                !!allExchangePurgeError
+                  ? `Failed to purge cached data for all the exchanges: ${allExchangePurgeError}`
+                  : null
+              "
+              @click="confirmAllExchangePurge = true"
+            >
+              Purge All Exchanges Cache
+            </status-button>
+            <h3>Ethereum Transactions</h3>
+            <status-button
+              class="data-management__purge-transactions"
+              tooltip="Purge the cached ethereum data"
+              :success-message="
+                transactionPurgeSuccess
+                  ? 'Ethereum transaction data were successfully purged.'
+                  : null
+              "
+              :error-message="
+                !!transactionPurgeError
+                  ? `Failed to purge cached ethereum transaction data: ${transactionPurgeError}`
+                  : null
+              "
+              @click="confirmTransactionPurge = true"
+            >
+              Purge transaction cache
+            </status-button>
+          </v-card-text>
+        </v-form>
+      </v-card>
+      <confirm-dialog
+        :display="confirmExchangePurge"
+        title="Purge cached exchange data"
+        :message="`Are you sure you want to purge the cached data for ${exchange}?`"
+        @confirm="purgeExchangeData()"
+        @cancel="confirmExchangePurge = false"
+      />
+      <confirm-dialog
+        :display="confirmAllExchangePurge"
+        title="Purge all cached exchange data"
+        :message="`Are you sure you want to purge the cached data for all the exchanges?`"
+        @confirm="purgeAllExchangeData()"
+        @cancel="confirmAllExchangePurge = false"
+      />
+      <confirm-dialog
+        :display="confirmTransactionPurge"
+        title="Purge ethereum transaction data"
+        :message="`Are you sure you want to purge the cached data for ethereum transactions?`"
+        @confirm="purgeEthereumTransactions()"
+        @cancel="confirmTransactionPurge = false"
```

Comment:
Could make this a bit DRYer by reducing the three confirmDialogs to 1 and where the component parameters take props? Opening/closing the dialog is done by a common on-click func that sets the relevant props and opens the dialog? The slight complexity is in the tailored `@confirm` and `@cancel` calls but that can be achieved by two methods:
Calling common funcs for confirm and cancel respecitvely where the scope of the func is an argument to the call, e.g. `purgeTransactions('ethereum')` and then handled with a switch in the func, or with something like this which allows you to do dynamic func calls:
https://jsfiddle.net/Roland1993/eywraw8t/418643/